### PR TITLE
Brokers can set status attribute on Inquiries

### PR DESCRIPTION
**Pull request by christerforsgren91**
PT https://www.pivotaltracker.com/story/show/178434414

### User story
```
As a broker
in order to keep track of my clients
I want to be able to change there current status
```

### Changes proposed
- Broker can update inquiries

### CODE REVIEW

File: app/controllers/api/inquiries_controller.rb [hunk 1]
```diff
@@ -2,10 +2,11 @@ class Api::InquiriesController < ApplicationController
...
     inquiry = Inquiry.create(inquiry_params)
 
-    if inquiry.persisted? 
+    if inquiry.persisted?
       render json: { message: 'Thanks for your answers! We\'ll be in touch' }
-    else 
-      render json: { error_message: 'Unfortunately, we had a small issue processing your request. Would you please try again?' }, status: 422
+    else
+      render json: { error_message: 'Unfortunately, we had a small issue processing your request. Would you please try again?' },
+             status: 422
```
Comment (lukeperrin10) [inline app/controllers/api/inquiries_controller.rb:9]: Formatting issue?

File: spec/requests/api/provider_can_update_inquiry_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    it 'is expected to respond with a message' do
+      expect(response_json['message']).to eq 'Inquiry has been updated'
+    end
+
+    it 'is expected to set inquiry status to started' do
+      expect(response_json['inquiry']['inquiry_status']).to eq 'started'
+    end
+  end
+end
```
Comment (lukeperrin10) [inline spec/requests/api/provider_can_update_inquiry_spec.rb]: No sad path?

File: spec/requests/api/provider_can_update_inquiry_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    it 'is expected to respond with a message' do
+      expect(response_json['message']).to eq 'Inquiry has been updated'
+    end
+
+    it 'is expected to set inquiry status to started' do
+      expect(response_json['inquiry']['inquiry_status']).to eq 'started'
+    end
+  end
```
Comment (LucasKnudsen) [inline spec/requests/api/provider_can_update_inquiry_spec.rb:22]: What's the reason that we're sending back the Inquiry? :)

**Review by LucasKnudsen — APPROVED** [review]

**Review by lukeperrin10 — CHANGES_REQUESTED** [review]

**Review by arunbhalli — APPROVED** [review]

File: app/controllers/api/inquiries_controller.rb (hunk 1 above)
Comment (1CIM) [reply app/controllers/api/inquiries_controller.rb:9]: seems like it 🤷‍♂️

File: spec/requests/api/provider_can_update_inquiry_spec.rb (hunk 2 above)
Comment (1CIM) [reply spec/requests/api/provider_can_update_inquiry_spec.rb]: nope no sad path

**Review by lukeperrin10 — APPROVED** [review]

File: spec/requests/api/provider_can_update_inquiry_spec.rb (hunk 2 above)
Comment (1CIM) [reply spec/requests/api/provider_can_update_inquiry_spec.rb]: now it dose

File: app/controllers/api/inquiries_controller.rb [hunk 4]
```diff
@@ -14,9 +15,18 @@ def index
     render json: inquiries, each_serializer: InquiriesIndexSerializer
   end
 
+  def update
+    inquiry = Inquiry.find(params[:id])
+    update_inquiry = inquiry.update(inquiry_status: params[:form_data][:inquiry_status])
```
Comment (oliverochman) [inline app/controllers/api/inquiries_controller.rb:20]: why do u save this in an variable?

File: db/migrate/20210607182009_add_status_to_inquiries.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToInquiries < ActiveRecord::Migration[6.1]
+  def change
+    add_column :inquiries, :inquiry_status, :integer
```
Comment (oliverochman) [inline db/migrate/20210607182009_add_status_to_inquiries.rb:3]: you should have an default value to this enum

File: spec/requests/api/provider_can_update_inquiry_spec.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+RSpec.describe 'PUT /api/inquiries/:id', type: :request do
+  let!(:inquiry) { create(:inquiry, inquiry_status: 'pending') }
```
Comment (oliverochman) [inline spec/requests/api/provider_can_update_inquiry_spec.rb:2]: unnecessary use of `let!` , you are using the variable in the test

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Pull request merged**